### DISCUSSION TITLE
Is there a way to call powershell runspace on .net7 NativeAOT?

### DISCUSSION THREAD

**Discussion by kasini3000** [PowerShell/PowerShell#17025] [General]
I want to use powershell to develop commercial software, using runspace and NativeAOT technology, 
can I achieve cross-platform green software?  Just like the golang command line program.

Assuming the name of the software is [foo], 
I hope that it can run without the linux version of powershell, 
can also be called by .so library in linux version of powershell.

----

Today, we are glad to release .NET 7 Preview 2. progress moving NativeAOT from experimental status into the runtime, 

https://devblogs.microsoft.com/dotnet/announcing-dotnet-7-preview-2/

**Comment by SeeminglyScience** [comment]
PowerShell is pretty heavily reflection dependent. I'm unsure how one could get PowerShell to work in a meaningful way inside of a trimmed application, especially one AOT compiled.

**Reply by kasini3000** [reply]
> PowerShell is pretty heavily reflection dependent.

Sounds bad. That is to say the powershell sdk is about to block aot?

**Reply by SeeminglyScience** [reply]
> Sounds bad. That is to say the powershell sdk is about to block aot?

I'm not sure what you mean there. It's definitely not going to block the feature from shipping, but you probably won't be able to utilize the PowerShell SDK in projects that are native AOT compiled.

To be clear though, this is just speculation.

**Reply by thgla** [reply]
Old discussion I know, but I figured it's worth to bring this up again - it still isn't supported, is it? I have a .NET 8 app that I tried compiling with Native AOT, and it does raise errors during runtime. I presume there's no way for this to get fixed without big rewrites in the SDK, so not likely we can use Native AOT anytime soon.

**Reply by fMichaleczek** [reply]
@thgla PowerShell is incompatible with NativeAoT by design. It's incompatible with trimming required by NativeAoT. Also NativeAoT means no library support, no Com+ support,  no assembly and no binary module. It will never happen.
